src/Core/functions.php

```diff
@@ -192,7 +192,7 @@ function pj($var)
      * @return string|bool|null Environment variable setting.
      * @link https://book.cakephp.org/4/en/core-libraries/global-constants-and-functions.html#env
      */
-    function env(string $key, $default = null)
+    function env(string $key, string|bool|null $default = null): string|bool|null
```

Comment:
I hope someone isn't setting integers.

Reply:
We could set it to mixed since the $default doesn't comply with getenv() types.

Reply:
IMO this method should only act as an abstraction to avoid modifying the actual environment vars and be complaint with `getenv()`. But that's a separate discussion.